server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -573,4 +439,166 @@ public TombstoneDocSupplier getTombstoneDocSupplier() {
     public TranslogDeletionPolicyFactory getCustomTranslogDeletionPolicyFactory() {
         return translogDeletionPolicyFactory;
     }
+
+    /**
+     * Builder for EngineConfig class
+     *
+     * @opensearch.internal
+     */
+    public static class Builder {
```

Comment:
Will this Builder make certain ctor parameters optional? How do we know we don't miss out on all mandatory ones

Reply:
The `build` method still calls the constructor (which is private now) so all the validations on parameters remain same.

Reply:
Agree on the validation but the notion of what are mandatory parameters and what are optional is diluted. Previously everything was mandatory, but is there something we could do to ease that and make this of the type EngineConfig#builder(mandatory_params).optionalParam(param).build(this)